src/components/tables/tests/TableRowReducers.spec.ts

```diff
@@ -99,36 +99,124 @@ describe('Tables', () => {
+      let oldState: ITableRowState[];
+      let openValue: boolean;
+
+      beforeEach(() => {
+        openValue = false;
+        oldState = [
+          { id: 'row2', opened: openValue, selected: false },
+          { id: 'row3', opened: openValue, selected: false },
+          { id: 'row1', opened: openValue, selected: false },
+        ];
+      });
+
+      it('should toggle the open property if the action is "TOGGLE_ROW" and isCollapsible is true', () => {
+        const action: IReduxAction<ITableRowActionPayload> = {
+          type: TableRowActions.toggle,
+          payload: {
+            id: 'row1',
+            isCollapsible: true,
+          }
+        };
+        let collapsibleRowsState: ITableRowState[] = tableRowsReducer(oldState, action);
+
+        expect(collapsibleRowsState.length).toBe(oldState.length);
+        expect(collapsibleRowsState.filter(row => row.id === action.payload.id)[0].opened).toBe(!openValue);
+        expect(collapsibleRowsState.filter(row => row.id !== action.payload.id)[0].opened).toBe(openValue);
+
+        collapsibleRowsState = tableRowsReducer(collapsibleRowsState, action);
+
+        expect(collapsibleRowsState.filter(row => row.id === action.payload.id)[0].opened).toBe(openValue);
+        expect(collapsibleRowsState.filter(row => row.id !== action.payload.id)[0].opened).toBe(openValue);
+      });
+
+      it('should leave the open property to a falsy value if the action is "TOGGLE_ROW" and isCollapsible is undefined', () => {
+        const action: IReduxAction<ITableRowActionPayload> = {
+          type: TableRowActions.toggle,
+          payload: {
+            id: 'row1',
+          }
+        };
+        let collapsibleRowsState: ITableRowState[] = tableRowsReducer(oldState, action);
+
+        expect(collapsibleRowsState.length).toBe(oldState.length);
+        expect(collapsibleRowsState.filter(row => row.id === action.payload.id)[0].opened).toBeUndefined();
+        expect(collapsibleRowsState.filter(row => row.id !== action.payload.id)[0].opened).toBeUndefined();
+
+        collapsibleRowsState = tableRowsReducer(collapsibleRowsState, action);
+
+        expect(collapsibleRowsState.filter(row => row.id === action.payload.id)[0].opened).toBeUndefined();
+        expect(collapsibleRowsState.filter(row => row.id !== action.payload.id)[0].opened).toBeUndefined();
+      });
+
+      it('should leave the open property to a falsy value if the action is "TOGGLE_ROW" and isCollapsible is false', () => {
+        const action: IReduxAction<ITableRowActionPayload> = {
+          type: TableRowActions.toggle,
+          payload: {
+            id: 'row1',
+            isCollapsible: false,
+          }
+        };
+        let collapsibleRowsState: ITableRowState[] = tableRowsReducer(oldState, action);
+
+        expect(collapsibleRowsState.length).toBe(oldState.length);
+        expect(collapsibleRowsState.filter(row => row.id === action.payload.id)[0].opened).toBeFalsy();
```

Comment:
We usually use `toBe(false);` instead of falsy (which treats 0, '', etc as falsy values)

Reply:
yes this was the test I wanted to make here, but with your suggested changes `opened` will always be false, it could have been undefined with the current implementation.

Reply:
I think it's easier to handle in the components if the value is either true or false and always defined